mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -467,7 +468,11 @@ ContiguousMergesMap findContiguousGroups(ArrayAttr transforms,
           thisDimToMerge[lowerDims[i]] = {transformMap, transform, i, false};
         }
         break;
+      // AddDim drops dimensions down a hole, while ConstDim conjures them
+      // from nowhere. In either case, there is no merge that can be associated
+      // with them.
       case TransformType::AddDim:
+      case TransformType::ConstDim:
```

Comment:
Shouldn't that also behave like a broadcast? I mean what happens if I have:
`Merge{8,3} -> ConstDim{1,5} -> Unmerge{8,5,3}` 
Now the Unmerge "thinks" that dim 0 and 2 are separate, while it's only an illusion

Reply:
After a chat we agreed that:
- If we add a unit dimension, the case is covered already
- If we add a non-unit dimension it should not work (and indeed it won't work)